### PR TITLE
WIP: update build

### PR DESCRIPTION
**Pull request by matkoch**

### CODE REVIEW

File: nukebuild/Build.cs [hunk 1]
```diff
@@ -93,29 +96,24 @@ void ExecWait(string preamble, string command, string args)
...
+            .SetVerbosity(MSBuildVerbosity.Minimal)
+            .Apply(configurator));
     }
+
     Target Clean => _ => _.Executes(() =>
     {
-        DeleteDirectories(Parameters.BuildDirs);
-        EnsureCleanDirectories(Parameters.BuildDirs);
+        Parameters.BuildDirs.ForEach(DeleteDirectory);
+        Parameters.BuildDirs.ForEach(DeleteDirectory);
```
Comment (kekekeks) [inline nukebuild/Build.cs]: EnsureCleanDirectories was ensuring that directories exist, right?

File: nukebuild/Build.cs [hunk 2]
```diff
@@ -134,97 +132,84 @@ void ExecWait(string preamble, string command, string args)
...
                 );
         });
-    
-    void RunCoreTest(string project)
+
+    void RunCoreTest(string projectName)
     {
-        if(!project.EndsWith(".csproj"))
-            project = System.IO.Path.Combine(project, System.IO.Path.GetFileName(project)+".csproj");
-        Information("Running tests from " + project);
```
Comment (kekekeks) [inline nukebuild/Build.cs]: Could we keep that line?

File: nukebuild/Build.cs [hunk 3]
```diff
@@ -134,97 +132,84 @@ void ExecWait(string preamble, string command, string args)
...
-        if (targets != null)
-            frameworks = targets.Value.Split(';').Where(f => !string.IsNullOrWhiteSpace(f)).ToList();
-        else 
-            frameworks = new List<string> {xdoc.Root.Descendants("TargetFramework").First().Value};
-        
-        foreach(var fw in frameworks)
+        Information($"Running tests from {projectName}");
+        var project = Solution.GetProject(projectName).NotNull("project != null");
+
+        foreach (var fw in project.GetTargetFrameworks())
```
Comment (kekekeks) [inline nukebuild/Build.cs]: Does it use MSBuild evaluation or parses stuff manually? We've seen issues with MSBuild on not-quite-supported OSes

File: nukebuild/_build.csproj [hunk 4]
```diff
@@ -2,15 +2,15 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
```
Comment (kekekeks) [inline nukebuild/_build.csproj]: This TFM was failing for me on several machines with .NET Core SDK 3.1 installed. Could we use `netcoreapp3.1` instead?

File: nukebuild/Build.cs (hunk 3 above)
Comment (matkoch) [reply nukebuild/Build.cs]: It uses MSBuild for evaluation. Could you try on those _not-quite-supported_ environments?

File: nukebuild/_build.csproj (hunk 4 above)
Comment (matkoch) [reply nukebuild/_build.csproj]: I guess so, let me try.

**Comment by grokys** [comment]
@matkoch what's the progress on this? The OSX build seems to be failing because the wrong version of .NET core is being installed?

**Comment by Gillibald** [comment]
https://github.com/AvaloniaUI/Avalonia/blob/master/azure-pipelines.yml#L41

**Comment by grokys** [comment]
@matkoch i've merged the changes in this PR into #3571, so going to close this. Thanks for the help, it's greatly appreciated!

**Pull request closed without merging**